apic_ml2/neutron/tests/unit/ml2/drivers/cisco/apic/test_network_constraints.py

```diff
@@ -0,0 +1,203 @@
+            return (def_scope, cons)
+
+    def setUp(self):
+        super(TestNetworkConstraints, self).setUp()
+        self.net_cons = anc.NetworkConstraints(self.MockSource())
+
+    def test_no_source(self):
+        net_cons = anc.NetworkConstraints(None)
+        self.assertEqual(
+            None,
+            net_cons.get_subnet_scope('t1', 'net1', '10.10.10.0/28'))
+        self.assertEqual(
+            None,
+            net_cons.get_subnet_scope('t1', 'net1', '0.0.0.0/0'))
+
+    def test_subnet_scope(self):
+        self.assertEqual(
+            anc.SCOPE_PUBLIC,
+            self.net_cons.get_subnet_scope('t1', 'net1', '10.10.10.0/28'))
+        self.assertEqual(
+            anc.SCOPE_PUBLIC,
+            self.net_cons.get_subnet_scope('t1', 'net1', '10.10.20.0/28'))
+
+        self.assertEqual(
+            anc.SCOPE_PRIVATE,
+            self.net_cons.get_subnet_scope('t1', 'net1', '20.10.10.1/28'))
+        self.assertEqual(
+            anc.SCOPE_PRIVATE,
+            self.net_cons.get_subnet_scope('t1', 'net1', '20.10.20.1/28'))
+
+        self.assertEqual(
+            anc.SCOPE_DENY,
+            self.net_cons.get_subnet_scope('t1', 'net1', '30.10.10.1/28'))
+        self.assertEqual(
+            anc.SCOPE_DENY,
+            self.net_cons.get_subnet_scope('t1', 'net1', '30.10.20.1/28'))
+
+    def test_network_default_subnet_scope(self):
+        self.assertEqual(
+            anc.SCOPE_PRIVATE,
+            self.net_cons.get_subnet_scope('t1', 'net1', '10.10.20.0/23'))
+        self.assertEqual(
+            anc.SCOPE_PRIVATE,
+            self.net_cons.get_subnet_scope('t1', 'net1', '40.10.10.1/28'))
+        self.assertEqual(
+            anc.SCOPE_DENY,
+            self.net_cons.get_subnet_scope('t1', 'net2', '10.10.10.1/28'))
+
+    def test_global_default_subnet_scope(self):
+        self.assertEqual(
+            anc.SCOPE_PUBLIC,
+            self.net_cons.get_subnet_scope('t1', 'net3', '10.10.10.1/24'))
+        self.assertEqual(
+            anc.SCOPE_PUBLIC,
+            self.net_cons.get_subnet_scope('t1', 'net4', '20.10.10.1/24'))
+        self.assertEqual(
+            anc.SCOPE_PUBLIC,
+            self.net_cons.get_subnet_scope('foo', 'bar', '10.10.10.1/24'))
+
+    def test_overlapping_subnet_scope(self):
+        self.assertEqual(
+            anc.SCOPE_DENY,
+            self.net_cons.get_subnet_scope('t2', 'net1', '5.5.10.128/25'))
+        self.assertEqual(
+            anc.SCOPE_PRIVATE,
+            self.net_cons.get_subnet_scope('t2', 'net2', '5.5.10.128/25'))
+        self.assertEqual(
+            anc.SCOPE_DENY,
+            self.net_cons.get_subnet_scope('t2', 'net3', '5.5.10.128/25'))
+        self.assertEqual(
+            anc.SCOPE_DENY,
+            self.net_cons.get_subnet_scope('t2', 'net4', '5.5.10.128/25'))
+
+
+class TestConfigFileSource(base.BaseTestCase):
+    def setUp(self):
+        super(TestConfigFileSource, self).setUp()
+        self.file_data = """
+[DEFAULT]
```

Comment:
Please also add an example file under the etc/ folder, with comments and examples to highlight the most common usage. Users can start from there to build their own net-conf file